compiler/runtime/OMRCodeMetaData.hpp

```diff
@@ -0,0 +1,118 @@
+/*******************************************************************************
+ *
+ * (c) Copyright IBM Corp. 2017, 2017
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ *******************************************************************************/
+
+#ifndef OMR_CODE_METADATA_INCL
+#define OMR_CODE_METADATA_INCL
+
+/*
+ * The following #define(s) and typedef(s) must appear before any #includes in this file
+ */
+#ifndef OMR_CODE_METADATA_CONNECTOR
+#define OMR_CODE_METADATA_CONNECTOR
+namespace OMR { struct CodeMetaData; }
+namespace OMR { typedef OMR::CodeMetaData CodeMetaDataConnector; }
+#endif
+
+#include "env/jittypes.h"         // for uintptrj_t
+#include "infra/Annotations.hpp"  // for OMR_EXTENSIBLE
+
+namespace TR { class CodeMetaData; }
+namespace TR { class Compilation; }
+
+/**
+ * CodeMetaData contains meta data information about a compilation unit 
+ * such as a method, a trace, etc.
+ */
+
+namespace OMR
+{
+
+class OMR_EXTENSIBLE CodeMetaData
+   {
+   public:
+
+   TR::CodeMetaData *self();
+
+   /**
+    * @brief Constructor method to create meta data for a compilation unit.
+    */
+   CodeMetaData(TR::Compilation *comp);
+
+   /**
+    * @brief Returns address of allocated code memory within a code
+    * cache for a compilation unit.
+    */
+   uintptrj_t codeAllocStart() { return _codeAllocStart; }
+
+   /**
+    * @brief Returns size of allocated code memory within a code cache 
+    * for a compilation unit.
+    *
+    * This requires compiled code segments, including snippets to be 
+    * laid out contiguously.
+    */
+   uint32_t codeAllocSize() { return _codeAllocSize; }
+
+   /**
+    * @brief Returns start PC address of compiled code for a
+    * compilation unit when invoked from interpreter.
+    *
+    * Interpreter entry PC may preceed compiled entry PC and may point 
+    * to code necessary for proper execution if invoked from interpreter, 
```

Comment:
* proper execution of this method
* from an interpreter
* For example, it may need to marshall method arguments from an interpreter linkage to a compiled method linkage.